ci/webhook-receiver.py

```diff
@@ -51,16 +84,40 @@ def filter_to_prs_and_pr_comments(json):
 
         return {
             "id": number,
+            "repo": repo_name,
+            "user": user_name,
             "body": comment_txt
         }
 
     return None
 
 @app.route(EVENT_URL, methods=['POST'])
 def init_ci_pipeline():
+    
+    if not verify_request_ip(request):
```

Comment:
When a server has to handle authN/authZ, it should maintain audit logs and error logs.  These logs are not to be exposed to github, but they are to be maintained internally.

The ideal way to go about this is to maintain some kind of "request context" that holds request specific info such as the github username, incoming ip, unique fields to identify the caller, and something to identify what gh op triggered the call (pr, issue, etc).  this ctx should be passed throughout the ci rig for any call.  

For every failure, the values from the req ctx and the failure message should be logged in the error log. 

authN/authZ failures, for example, all of these `verify_request_*` calls and the check to see if the user is in the whitelist, should be logged in an audit log.  these logs should contain alerts about who failed to log in and who is not authorized, and why they failed to log in.